### PR TITLE
[octomap] update to 1.10.0

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/octomap/portfile.cmake [hunk 1]
```diff
@@ -4,11 +4,11 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO OctoMap/octomap
     REF "v${VERSION}"
-    SHA512 60afeecc36a190f136dcbe33cb9cd6c06c16233988b383b0b010f65f81e6a3630b55902c5b5ad756ac35dee4c4ec26ec5722d6bd9b8e079f70b7d286293c518e
+    SHA512 1cbee4f6b3569587986774447ad9ec4190f597310c4d6865ffa7cd8865ece2492e4a42fa369b633d9d7a9da782560d49deaa62a18601ea4f56396bdf1a6a5f52
     HEAD_REF master
```
Comment (Cheney-W) [inline ports/octomap/portfile.cmake:8]:
```suggestion
    HEAD_REF devel
```

File: ports/octomap/fix-isnan.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+--- a/octomap/src/compare_octrees.cpp
++++ b/octomap/src/compare_octrees.cpp
+@@ -39,10 +39,6 @@
+ #include <list>
+ #include <cmath>
+ 
+-#ifdef _MSC_VER // fix missing isnan for VC++
+-#define isnan(x) _isnan(x)  
+-#endif
+-
```
Comment (Cheney-W) [inline ports/octomap/fix-isnan.patch:12]: Does the upstream agree to your proposed approach to fix the below error occurring during builds on Windows?
```
octomap\src\compare_octrees.cpp(136): error C2039: '_isnan': is not a member of 'std'
```

Comment (BillyONeal) [reply ports/octomap/fix-isnan.patch:12]: It looks like this is undoing upstream's workaround to fix that error..

Comment (jiayuehua) [reply ports/octomap/fix-isnan.patch:12]: I open a pull request on upstream OctoMap/octomap#417

Comment (jiayuehua) [reply ports/octomap/fix-isnan.patch]: https://github.com/OctoMap/octomap/pull/417 has be merged on upstream repo. I have just update the patch.

**Review by Cheney-W — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update and following up with upstream! :)

**Pull request merged**